### DISCUSSION TITLE
How to handle BasicAuth with swagger-typescript-api?

### DISCUSSION THREAD

**Discussion by linkoln-1** [acacode/swagger-typescript-api#917] [Q&A]
Hi,

I'm trying to generate TypeScript types from a Swagger endpoint protected by BasicAuth using the swagger-typescript-api package. However, I couldn't find a way to pass the Authorization header or BasicAuth credentials in the command. Is there a supported way to provide authentication (either via headers or another method) to access the protected API? Any guidance would be appreciated!

Thanks!

**Comment by reite** [comment]
When initializing the generated class there is a parameter `securityWorker` that can be used for this. 

```typescript
import { Api } from "...";

const api = new Api({
  securityWorker: () => ({
    headers: {
      Authorization: `Bearer ${token}`
    },
  }),
});
```

**Reply by RubenFricke** [reply]
Can this also be accomplished using the terminal only (or via npm, but I prefer to just use one command to access my restricted swagger environment)?

**Comment by bibullo** [comment]
the way I did it was through their `generateApi` method.

```
import { generateApi } from 'swagger-typescript-api';

generateApi({
  name: 'generated.types.ts',
  output: path.resolve(process.cwd(), './types'),
  url: '...',
  authorizationToken: `Basic ${btoa(`${username:password}`)}`,
  generateClient: false,
});
```

then to get the required `username` and `password` credentials I setup 2 possible ways of providing it, NPM script params or node env file.

```
const username = process.env.npm_config_username || process.env.API_USERNAME;
const password = process.env.npm_config_password || process.env.API_PASSWORD;
```

with the following npm script: `"generate-types": "node --env-file=.env generate-types.mjs"`

so finally it's possible to run either: 
- `npm run generate-types --username=... --password=...`
 or 
- `npm run generate-types` with a `.env` file setup with the credentials
